spec/patterns/firewalls_spec.rb

```diff
@@ -2,135 +2,254 @@
-
-    subject { grok_match(pattern106023, value) }
+    let(:message) { 'Deny protocol 103 src outside:192.168.1.1/50240 dst inside:192.168.1.2/23 by access-group "S_OUTSIDE_TO_INSIDE" [0x54c7fa80, 0x0]' }
 
     it 'should break the message up into fields' do
       expect(subject['action']).to eq('Deny')
       expect(subject['src_interface']).to eq('outside')
       expect(subject['dst_interface']).to eq('inside')
-      expect(subject['protocol']).to eq('tcp')
+      expect(subject['protocol']).to eq('103')
       expect(subject['src_ip']).to eq('192.168.1.1')
       expect(subject['dst_ip']).to eq('192.168.1.2')
       expect(subject['policy_id']).to eq('S_OUTSIDE_TO_INSIDE')
     end
   end
 
-  context "parsing a 106023 message with a protocol number" do
 
-    let(:value) { 'Deny protocol 103 src outside:192.168.1.1/50240 dst inside:192.168.1.2/23 by access-group "S_OUTSIDE_TO_INSIDE" [0x54c7fa80, 0x0]' }
+  context "a message with a hostname" do
 
-    subject { grok_match(pattern106023, value) }
+    let(:message) { 'Deny tcp src outside:192.168.1.1/50240 dst inside:www.example.com/23 by access-group "S_OUTSIDE_TO_INSIDE" [0x54c7fa80, 0x0]' }
 
     it 'should break the message up into fields' do
       expect(subject['action']).to eq('Deny')
       expect(subject['src_interface']).to eq('outside')
       expect(subject['dst_interface']).to eq('inside')
-      expect(subject['protocol']).to eq('103')
+      expect(subject['protocol']).to eq('tcp')
       expect(subject['src_ip']).to eq('192.168.1.1')
-      expect(subject['dst_ip']).to eq('192.168.1.2')
+      expect(subject['dst_ip']).to eq('www.example.com')
       expect(subject['policy_id']).to eq('S_OUTSIDE_TO_INSIDE')
     end
   end
 
-  context "parsing a 106023 message with a hostname" do
+end
 
-    let(:value) { 'Deny tcp src outside:192.168.1.1/50240 dst inside:www.example.com/23 by access-group "S_OUTSIDE_TO_INSIDE" [0x54c7fa80, 0x0]' }
 
-    subject { grok_match(pattern106023, value) }
+describe_pattern 'SFW2', ['legacy', 'ecs-v1'] do
 
-    it 'should break the message up into fields' do
-      expect(subject['action']).to eq('Deny')
-      expect(subject['src_interface']).to eq('outside')
-      expect(subject['dst_interface']).to eq('inside')
-      expect(subject['protocol']).to eq('tcp')
-      expect(subject['src_ip']).to eq('192.168.1.1')
-      expect(subject['dst_ip']).to eq('www.example.com')
-      expect(subject['policy_id']).to eq('S_OUTSIDE_TO_INSIDE')
+  let(:message) do
+    "Jan 29 00:00:28 myth kernel: SFW2-INext-DROP-DEFLT IN=ppp0 OUT= MAC= SRC=24.64.208.134 DST=216.58.112.55 LEN=512 TOS=0x00 PREC=0x00 TTL=70 ID=55012 PROTO=UDP SPT=24128 DPT=1026 LEN=492"
+  end
+
+  it 'matches' do
+    # NOTE: we do not match the second LEN=492 which is the length of the wrapped (UDP in this case) packet
+    # iptables.length (IP packet length) 512 = 492 (UDP/TCP packet length) + 20 (IPv4 header length = 20 bytes)
```

Comment:
as noted here and also visible on the PR's description example we only capture the first `LEN=...` field as
`iptables.length` (IP packet length), second `LEN` represent (UDP/TCP packet length)`iptables.length - 20` (where `20` stands for IPv4 header length in bytes). does it make sense to capture the other as well, if so I was a bit lost on naming (`iptables.payload_length`) ?